### PR TITLE
[CAP-400] - implement assign task component

### PR DESCRIPTION
**Pull request by CaoSon1909**

### CODE REVIEW

File: jobfair-frontend/src/containers/AssignTask/AssignTask.container.jsx [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+      setBoothData(jobFairBooth);
+      setShiftData(shiftData);
+    } catch (e) {
+      notification['error']({
+        message: 'Error when get employee data'
+      });
+    }
+  };
+
+  const getTableColumns = async (dayRange) => {
```
Comment (ZERO-94) [inline jobfair-frontend/src/containers/AssignTask/AssignTask.container.jsx:120]: column file ?

File: jobfair-frontend/src/containers/AssignTask/AssignTask.container.jsx [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+  //each row of the datasource comprise of employeeId, employee's info, assignments by date
+  //Ex: [
+  //  {
+  //    employeeId: "f996f062-c1d5-4fcc-a13c-8cae1e210d0d",
+  //    employee: {accountId: ....},
+  //    'Sat, Jun/4/2022': [{assignment1, assignment2}],
+  //    'Sun, Jun/5/2022': [{assignment1, assignment2}],
+  //  }
+  // ]
+  const getDataSource = async (staffAssignments, staffs, dayRange, shiftData) => {
```
Comment (ZERO-94) [inline jobfair-frontend/src/containers/AssignTask/AssignTask.container.jsx:163]: separate file ?

File: jobfair-frontend/src/containers/AssignTask/AssignTask.container.jsx [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+      const title = current.format('ddd, MMM/D/YYYY');
+      prev[title] = { beginTime: current.clone(), endTime: current.clone().endOf('day') };
+      return prev;
+    }, {});
+    return dayRange;
+  };
+
+  //get employee information and store in map with key is employeeId
+  //Ex: {123: {accountId:....}}
+  const getStaffList = async (assignments) => {
```
Comment (ZERO-94) [inline jobfair-frontend/src/containers/AssignTask/AssignTask.container.jsx:225]: separate file

File: jobfair-frontend/src/containers/AssignTask/AssignTask.container.jsx [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+      .map((assignment) => assignment.companyEmployee)
+      .reduce((prev, employee) => {
+        prev[employee.accountId] = employee;
+        return prev;
+      }, {});
+  };
+
+  //transform staffAssignments from list of assignment to map of assignments based on employeeId
+  //Ex: {3128aa05-fafa-4790-a8dd-219b6741f9d4: [{assignment1}, {assignment2}]
+  const getStaffAssignments = async (assignments, shiftData) => {
```
Comment (ZERO-94) [inline jobfair-frontend/src/containers/AssignTask/AssignTask.container.jsx:239]: separate file

File: jobfair-frontend/src/containers/AssignTask/AssignTask.container.jsx [hunk 5]
```diff
@@ -0,0 +1,490 @@
...
+    try {
+      await form.validateFields();
+      const values = form.getFieldsValue(true);
+      const data = tableData.dataSource.find((item) => item.employeeId === selectedCellData.employee.accountId);
+      const assignments = data[selectedCellData.date];
+      for (let i = 0; i < shiftData.length; i++) {
+        if (values[`shift-${i}`] === true) {
+          const assignment = assignments.find((assignment) => assignment.shift === i);
+          if (assignment === undefined) {
+            const result = {
```
Comment (ZERO-94) [inline jobfair-frontend/src/containers/AssignTask/AssignTask.container.jsx:294]: mapper

File: jobfair-frontend/src/containers/AssignTask/AssignTask.container.jsx [hunk 6]
```diff
@@ -0,0 +1,490 @@
...
+    }));
+  };
+
+  const onSaveChange = async () => {
+    //compare oldDataSource and newDataSource
+    const newAssignments = [];
+    const deletedAssignments = [];
+    const updatedAssignmens = [];
+
+    for (let i = 0; i < tableData.oldDataSource.length; i++) {
```
Comment (ZERO-94) [inline jobfair-frontend/src/containers/AssignTask/AssignTask.container.jsx:352]: separate file ?

File: jobfair-frontend/src/containers/AssignTask/AssignTask.container.jsx [hunk 7]
```diff
@@ -0,0 +1,490 @@
...
+      setForceRerender((prevState) => !prevState);
+      setHasChange(false);
+    } catch (e) {
+      notification['error']({
+        message: `Error occurred: ${e.response.data.message}`
+      });
+    }
+  };
+
+  const onFilterData = (values) => {
```
Comment (ZERO-94) [inline jobfair-frontend/src/containers/AssignTask/AssignTask.container.jsx:410]: separate file ?

**Review by ZERO-94 — CHANGES_REQUESTED** [review]

File: jobfair-frontend/src/components/forms/ScheduleJobFairForm/ScheduleJobFairForm.component.jsx [hunk 8]
```diff
@@ -7,35 +7,41 @@ import moment from 'moment';
 
 const { RangePicker } = DatePicker;
 const { Title } = Typography;
-const ScheduleJobFairFormComponent = ({ jobFairData, form, onFinish, onValueChange }) => {
+
+const mapJobFairDataToFormInitialValues = (jobFairData) => {
   const shift = jobFairData.shifts?.sort((a, b) => a.beginTime - b.beginTime);
   const startOfDate = moment().startOf('day').valueOf();
+  return {
```
Comment (ZERO-94) [inline jobfair-frontend/src/components/forms/ScheduleJobFairForm/ScheduleJobFairForm.component.jsx:14]: mapper ???

File: jobfair-frontend/src/containers/AssignTask/column/AssignTask.column.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
+import { AssignTaskCell } from '../../../components/customized-components/AssignTask/AssignTaskCell.component';
+import { Avatar, Space, Typography } from 'antd';
+import React from 'react';
+
+const { Text } = Typography;
+
+export const getTableColumns = async (dayRange, handleOpenAssignTaskModal) => {
```
Comment (ZERO-94) [inline jobfair-frontend/src/containers/AssignTask/column/AssignTask.column.js:7]: async ?

File: jobfair-frontend/src/containers/AssignTask/AssignTask.container.jsx [hunk 10]
```diff
@@ -281,6 +84,17 @@ const AssignTaskContainer = (props) => {
     }));
   };
 
+  const generateAssignment = (data, shiftData, shiftIndex, assignmentType) => ({
+    beginTime: tableData.dayRange[selectedCellData.date].beginTime.valueOf() + shiftData[shiftIndex].beginTime,
```
[the remark this answers is not in the document]
Comment (TienTruong2000) [reply jobfair-frontend/src/containers/AssignTask/AssignTask.container.jsx:88]: this is a mapper

File: jobfair-frontend/src/components/forms/ScheduleJobFairForm/ScheduleJobFairForm.component.jsx (hunk 8 above)
Comment (TienTruong2000) [reply jobfair-frontend/src/components/forms/ScheduleJobFairForm/ScheduleJobFairForm.component.jsx:14]: this is a mapper

File: jobfair-frontend/src/containers/OrganizeJobFair/OrganizeJobFair.container.jsx [hunk 11]
```diff
@@ -101,13 +102,26 @@ const OrganizeJobFairContainer = () => {
...
+        {
+          beginTime: Math.trunc((values.morningShift[0].valueOf() - startOfDate.valueOf()) / 1000) * 1000,
+          endTime: Math.trunc((values.morningShift[1].valueOf() - startOfDate.valueOf()) / 1000) * 1000
+        },
+        {
+          beginTime: Math.trunc((values.afternoonShift[0].valueOf() - startOfDate.valueOf()) / 1000) * 1000,
+          endTime: Math.trunc((values.afternoonShift[1].valueOf() - startOfDate.valueOf()) / 1000) * 1000
+        }
+      ];
       const body = {
```
[the remark this answers is not in the document]
Comment (ZERO-94) [reply jobfair-frontend/src/containers/OrganizeJobFair/OrganizeJobFair.container.jsx:112]: @TienTruong2000 
here

**Review by ZERO-94 — APPROVED** [review]

**Pull request merged**